exasol_integration_test_docker_environment/lib/docker/images/create/utils/file_directory_list_hasher.py

```diff
@@ -60,20 +70,36 @@ def __init__(self,
                                       use_relative_paths=use_relative_paths)
         self.file_content_hasher = FileContentHasher(hashfunc, blocksize)
 
-    def hash(self, files_and_directories: List[str]) -> bytes:
-        collected_directories = []
-        collected_files = []
+    def hash(self, files_and_directories: List[PathMapping]) -> bytes:
+        collected_directories: List[RelativePath] = list()
+        collected_files: List[RelativePath] = list()
+
         if not isinstance(files_and_directories, List):
             raise Exception("List with paths expected and not '%s' with type %s"
                             % (files_and_directories, type(files_and_directories)))
         for file_or_directory in files_and_directories:
-            if os.path.isdir(file_or_directory):
-                self.collect_files_and_directories(
-                    file_or_directory, collected_directories, collected_files)
-            elif os.path.isfile(file_or_directory):
-                collected_files.append(("", file_or_directory))
+            source = file_or_directory.source
+            dest = file_or_directory.destination
+            if not source.endswith(dest):
+                raise AssertionError("source and destination must match. "
+                                     "Pathname of destination must be a suffix of source. "
+                                     "Otherwise the root path cannot be determined")
+            # We calculate the root path based on the information in the mapping, examples:
+            # 1. PathMapping(dest="requirements.txt", source="requirements.txt") => root_path = "."
+            # 2. PathMapping(dest="test/requirements.txt", source="test/requirements.txt") => root_path = "."
+            # 3. PathMapping(dest="requirements.txt", source="test/requirements.txt") => root_path = "./test"
+            # 4. PathMapping(dest="requirements.txt", source="/tmp/tmp123/requirements.txt") => root_path = "/tmp/tmp123/"
+            root_path = Path(source.rstrip(dest))
```

Comment:
This looks wrong. We don't care if destination is part of source. What we want to do, is to replace source completely by destination. For example:

src: /home/user1/test
dest: mytest

And under src we have

/home/user1/test/file1
/home/user1/test/dir/file2

Then we want the following names for hashing 

mytest/file1
mytest/dir/file2

Reply:
fixed

Reply:
as discussed, with the new implementation we use the destination paths  names as input for hashing.